packages/documentsearch/test/documentsearchprovider.spec.ts

```diff
@@ -22,6 +22,23 @@ describe('documentsearch/genericsearchprovider', () => {
       widget.dispose();
     });
 
+    describe('#preserveCase()', () => {
+      it.each([
+        ['OLD_TEXT_1', 'new_text_1', 'NEW_TEXT_1'],
+        ['OLD_TEXT_1', 'NEW_TEXT_1', 'NEW_TEXT_1'],
+        ['old_text_1', 'new_text_1', 'new_text_1'],
+        ['old_text_1', 'NEW_TEXT_1', 'new_text_1'],
+        ['Old', 'new', 'New'],
+        ['Old', '𐐶ew', '𐐎ew']
```

Comment:
Maybe add (and implement) more cases like:

```
['OldText', 'new_text', 'NewText'],
['old_text', 'NewText', 'new_text']
```

Do we want this?

Reply:
Do you know what VS Code is doing?

It seems pretty complex to handle. I would go for moving forward with this proposal. It can always be improved latter.

Reply:
Yes, I can open a followup PR, this way we can merge this one which is rather uncontroversial and then focus on the specific extra cases judging if they are worth extra complexity.